cfme/tests/v2v/test_csv_import.py

```diff
@@ -0,0 +1,167 @@
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.generators import random_vm_name
+from cfme.utils.wait import wait_for
+
+pytestmark = [
+    pytest.mark.ignore_stream('5.8'),
+    pytest.mark.provider(
+        classes=[RHEVMProvider],
+        selector=ONE_PER_VERSION
+    ),
+    pytest.mark.provider(
+        classes=[VMwareProvider],
+        selector=ONE_PER_VERSION,
+        fixture_name='second_provider'
+    )
+]
+
+
+@pytest.fixture(scope="function")
+def ims_map(appliance, v2v_providers):
+    """Fixture to create infrastructure mapping"""
+    form_data = _form_data(v2v_providers[0], v2v_providers[1])
+    return appliance.collections.v2v_mappings.create(form_data)
+
+
+def ims_plan(appliance, ims_map, csv=None):
+    """Function to create migration plan and select csv import option"""
+    plan_name = "map_{}".format(fauxfactory.gen_alpha(10))
+    plan_obj = appliance.collections.v2v_plans
+    view = navigate_to(plan_obj, 'Add', wait_for_view=True)
+    view.general.fill({
+        'infra_map': ims_map.name,
+        'name': plan_name,
+        'description': fauxfactory.gen_alpha(20)
+    })
+    if not csv:
+        view.general.select_vm.select("Import a CSV file with a list of VMs to be migrated")
+        view.next_btn.click()
+    return view
+
+
+@pytest.fixture(scope="function")
+def valid_vm(appliance, ims_map):
+    """Fixture to get valid vm name from discovery"""
+    plan_view = ims_plan(appliance, ims_map, csv=True)
+    plan_view.next_btn.click()
+    vm_name = [row.vm_name.read() for row in plan_view.vms.table.rows()][0]
+    plan_view.cancel_btn.click()
+    return vm_name
+
+
+@pytest.fixture(scope="function")
+def archived_vm(appliance, second_provider):
+    """Fixture to create archived vm"""
+    vm_obj = appliance.collections.infra_vms.instantiate(
+        random_vm_name(context='v2v-auto'), second_provider)
+    if not second_provider.mgmt.does_vm_exist(vm_obj.name):
+        vm_obj.create_on_provider(find_in_cfme=True, allow_skip="default")
+    vm_obj.mgmt.delete()
+    vm_obj.wait_for_vm_state_change(desired_state='archived', timeout=900,
+                                    from_details=False, from_any_provider=True)
+    return vm_obj.name
+
+
+def test_blank_csv(appliance, ims_map):
+    """Test csv with blank file"""
+    plan_view = ims_plan(appliance, ims_map)
+    temp_file = tempfile.NamedTemporaryFile(suffix='.csv')
+    plan_view.vms.hidden_field.fill(temp_file.name)
+    error_msg = "Error: Possibly a blank .CSV file"
+    if plan_view.vms.error_icon.is_displayed:
+        assert error_msg == plan_view.vms.error_text.text
+        plan_view.cancel_btn.click()
+
+
+def test_column_headers(appliance, ims_map):
+    """Test csv with unsupported column header"""
+    plan_view = ims_plan(appliance, ims_map)
+    temp_file = tempfile.NamedTemporaryFile(suffix='.csv')
+    with open(temp_file.name, 'w') as file:
```

Comment:
Required:
can you please use `f` or other names to open a file? `file` is `builtin` while playing with files means its `type` of `file`.